pkg/arvo/lib/group.hoon

```diff
@@ -0,0 +1,98 @@
+/-  *group, *metadata-store
+/+  store=group-store
+|_  =bowl:gall
+++  scry-for
+  |*  [=mold =path]
+  .^  mold
+    %gx
+    (scot %p our.bowl)
+    %group-store
+    (scot %da now.bowl)
+    (snoc `^path`path %noun)
+  ==
+::
+++  scry-group-path
+  |=  =path
+  %+  scry-for
+    (unit group)
+  [%groups path]
+::
+++  scry-group
+  |=  =group-id
+  %-  scry-group-path
+  (group-id:en-path:store group-id)
+::
+++  members
+  |=  =group-id
+  %-  members-from-path
+  (group-id:en-path:store group-id)
+::
+++  members-from-path
+  |=  =group-path
+  ^-  (set ship)
+  =-  members:(fall - *group)
+  (scry-group-path group-path)
+::
+++  is-permitted
+  |=  [=ship =group-path]
+  ^-  ?
+  =-  (~(has in -) ship)
+  (members-from-path group-path)
+::
+++  role-for-ship
+  |=  [=group-id =ship]
+  ^-  (unit role-tag)
+  =/  grp=(unit group)
+    (scry-group group-id)
+  ?~  grp  ~
+  =*  group   u.grp
+  =*  policy  policy.group
+  =*  tag-queries  tag-queries.group
+  =/  admins=(set ^ship)
+    (~(gut by tag-queries) %admin ~)
+  ?:  (~(has in admins) ship)
+    `%admin
+  =/  mods
+    (~(gut by tag-queries) %moderator ~)
+  ?:  (~(has in mods) ship)
+    `%moderator
+  =/  janitors
+    (~(gut by tag-queries) %janitor ~)
+  ?:  (~(has in janitors) ship)
+    `%janitor
+  ?:  (~(has in members.group) ship)
+    `%member
+  ~
+++  can-join-from-path
+  |=  [=path =ship]
+  %+  scry-for
+    ?
+  %+  welp
+    [%groups path]
+  /join/[(scot %p ship)]
+::
+++  can-join
+  |=  [=group-id =ship]
+  =-  ~&  -  -
+  %+  can-join-from-path
+    (group-id:en-path:store group-id)
+  ship
+++  is-managed-path
+  |=  =path
+  =/  contact
+    .^  (unit *)
+      %gx
+      (scot %p our.bowl)
+      %contact-store
+      (scot %da now.bowl)
+      (snoc `^path`[%contacts path] %noun)
+    ==
+  ?~  contact
+    %.n
+  %.y
+++  is-managed
```

Comment:
These are kinda weird. Scry into contact-store? Check for existence?

I mean, I get what it's doing, but a comment would be very useful here. And we probably don't want to depend on contact-store for this, anyway? Did we not consider including a `hidden=?` flag in groups?

Reply:
We'd be duplicating this information across stores then, which is part of what caused the whole permissions/groups mess in the first place. I'm not 100% opposed to it, just think we should minimise duplication across stores where possible.

Reply:
I think the mess there was that it was implicit in paths/identifiers, which was hard and unnecessary to track.

The root of that information would be what the _group_ was to be used for. You only ever care about this on the group level. A chat is just a chat, but it may be associated with a visible or hidden group. As such, you just have apps refer to the data here, and don't end up with duplication.

Does that sound about right?

Reply:
There should be no need for `is-managed` now that we have tagged-queries for groups.